source/extensions/transport_sockets/starttls/config.cc

```diff
@@ -16,37 +10,49 @@ namespace StartTls {
 Network::TransportSocketFactoryPtr DownstreamStartTlsSocketFactory::createTransportSocketFactory(
     const Protobuf::Message& message, Server::Configuration::TransportSocketFactoryContext& context,
     const std::vector<std::string>& server_names) {
-  const auto& outer_config = MessageUtil::downcastAndValidate<
-      const envoy::extensions::transport_sockets::starttls::v3::StartTlsConfig&>(
-      message, context.messageValidationVisitor());
+  const auto& outer_config = castConfig(message, context.messageValidationVisitor());
```

Comment:
The implementation of `castConfig` is just a one liner. I think it is totally OK to just use the `MessageUtil::downcastAndValidate` instead of calling a method.